### PR TITLE
feat: Avoid use the secret value in the DevOps scm verify API.

### LINKED ISSUE #2893: Avoid use the secret value in the DevOps scm verify API
**What would you like to be added**:

API: POST `devops.kubesphere.io/v1alpha2/scms/github/verify/`

Payload: `{accessToken: "github_plain_token"}`

Suggestion: change the payload to `{name: 'secret_name'}`

**Why is this needed**:

There are security issues if users can get a plain secret value. See also the above example:

We should never pass the token to an API. Instead, please pass an ID or name on it.

/area devops

## References
* https://github.com/kubesphere/ks-devops/issues/299

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

### What type of PR is this?

/kind feature

### What this PR does / why we need it:
Please see #2893

### Which issue(s) this PR fixes:

Fixes #2893

### Special notes for reviewers:
The request Payload screenshot like below:

image

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
hi @mangoGoForward thanks for your time. I'm sorry that I didn't update the description in time. But the new API was `/v1alpha3/scms/github/verify`

See the details from https://github.com/kubesphere/ks-devops/pull/439

**Comment by mangoGoForward** [comment]
> hi @mangoGoForward thanks for your time. I'm sorry that I didn't update the description in time. But the new API was `/v1alpha3/scms/github/verify`
> 
> See the details from [kubesphere/ks-devops#439](https://github.com/kubesphere/ks-devops/pull/439)
> 
> The old API does not support it yet. See the following screenshot:
> 
> image

I noticed that the API of `/v1alpha3/scms/github/verify` has two query parameters,

https://github.com/kubesphere/ks-devops/blob/d54a80ebd16c0df7b4698869babb17e41fec681e/pkg/kapis/devops/v1alpha3/register.go#L214-L215.

please correct me, there are two things:
1. update API version.
2. add this two parameters instead of `accessToken`.

But I'm confused that the API why would throw the error of `accessToken is required`?

**Comment by LinuxSuRen** [comment]
You got the right way.

The old API does not accept the parameter `name`. This is the reason why it will throw the error.

**Comment by mangoGoForward** [comment]
> ` [1ac41c4](/kubesphere/console/pull/2943/commits/1ac41c45c23a4aa21f57d3cf9ea0a55be5a8f7e8)`

Updated, please see 1ac41c4.

File: src/components/Forms/Pipelines/RepoSelect/GithubForm/index.jsx [hunk 1]
```diff
@@ -146,8 +146,9 @@ export default class GitHubForm extends React.Component {
 
     if (token) {
       const secretName = get(credentialDetail, 'name')
+      const secretNamespace = get(credentialDetail, 'namespace')
       await this.props.store
-        .putAccessName({ secretName, name, cluster, devops })
+        .putAccessName({ secretName, secretNamespace, name, cluster, devops })
```
Comment (LinuxSuRen) [inline src/components/Forms/Pipelines/RepoSelect/GithubForm/index.jsx:151]: do we still need the parameter `name`?

Comment (mangoGoForward) [reply src/components/Forms/Pipelines/RepoSelect/GithubForm/index.jsx:151]: Not need, and the `putAccessName ` function would not receive `name` parameter.

Comment (LinuxSuRen) [reply src/components/Forms/Pipelines/RepoSelect/GithubForm/index.jsx:151]: How about just keeping the necessary parameters?

Comment (mangoGoForward) [reply src/components/Forms/Pipelines/RepoSelect/GithubForm/index.jsx]: Done.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
I got some problems when I test this PR manually.

image

image

Sorry about I didn't provide clear information about the API. See https://github.com/kubesphere/ks-devops/pull/439

The API takes `secret` and `secretNamespace` as the query parameters instead of payload.

The second problem is that I didn't get any tips when there is an error code. It will make users feel confused.

Sorry again for the inconvenience. I'll try to fix it based on your PR.

**Comment by mangoGoForward** [comment]
> The API takes `secret` and `secretNamespace` as the query parameters instead of payload.

Sorry, I have not noticed that the use query parameters instead of payload, but the API method is `POST`,  may use query body better.

> The second problem is that I didn't get any tips when there is an error code. It will make users feel confused.

I will check.

**Comment by LinuxSuRen** [comment]
I found out the front-end still call the following API which expose the password

image

**Comment by mangoGoForward** [comment]
> I found out the front-end still call the following API which expose the password
> 
> image

Seems can hide in this API, we use `scmType`, secret name and namespaces to instead of `data.password` now.

**Comment by LinuxSuRen** [comment]
> Seems can hide in this API, we use scmType, secret name and namespaces to instead of data.password now.

It would be better.

**Comment by LinuxSuRen** [comment]
Give the upstream project [go-scm](https://github.com/drone/go-scm) does not support listing repositories from a user or organization for now. I guess this PR needs to be held.

/hold

**Comment by LinuxSuRen** [comment]
Please feel free to use the following image if you want to help to test this PR:
```
surenpi/ks-console:latest@sha256:449d6ffacf2ab3a9984b106cb93c028e152ca6ab508e6a6f1039167e6264b5ef
```

**Review by LinuxSuRen — APPROVED** [review]
I have tested it manually. It works well with [the last commit](https://github.com/kubesphere/console/pull/2943/commits/dc9e8851b8cc433575fd893f9bb65a95753aa2c5). See the following screenshot:

image

/lgtm
/approve

**Pull request merged**